File: apps/wallpad/t527_vt_config_test/app/src/main/java/com/t527/wav2vecdemo/BootReceiver.java
package com.t527.wav2vecdemo;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

public class BootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            Log.d("BootReceiver", "Boot completed, starting VadPipelineService");
            Intent serviceIntent = new Intent(context, VadPipelineService.class);
            context.startForegroundService(serviceIntent);
        }
    }
}

File: apps/wallpad/t527_smart_service/app/src/main/kotlin/com/t527/smart_service/BootReceiver.kt
package com.t527.smart_service

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == Intent.ACTION_BOOT_COMPLETED) {
            Log.d("BootReceiver", "Boot completed, starting VoiceAiService")
            val svcIntent = Intent(context, VoiceAiService::class.java).apply {
                action = VoiceAiService.ACTION_MIC_GRANTED
            }
            context.startForegroundService(svcIntent)
        }
    }
}
